cmd/extension.go

```diff
@@ -0,0 +1,100 @@
+package cmd
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/odpf/optimus/extension"
+	"github.com/odpf/salt/log"
+
+	"github.com/google/go-github/github"
+	cli "github.com/spf13/cobra"
+)
+
+func addExtensionCommand(cmd *cli.Command, l log.Logger) {
+	ctx := context.Background()
+	httpClient := &http.Client{}
+	githubClient := github.NewClient(nil)
+	manifest, err := extension.LoadManifest()
+	if err != nil {
+		panic(err)
+	}
+	reservedCommands := getUsedCommands(cmd)
+	extension := extension.NewExtension(ctx, manifest, githubClient, httpClient, reservedCommands...)
+
+	cmd.AddCommand(extensionCommand(extension.Install, l))
+	commands := generateCommands(manifest, extension.Run)
+	for _, c := range commands {
+		cmd.AddCommand(c)
+	}
+}
+
+func generateCommands(manifest *extension.Manifest, execFn func(string, []string) error) []*cli.Command {
+	output := make([]*cli.Command, len(manifest.Metadatas))
+	for i, metadata := range manifest.Metadatas {
+		c := &cli.Command{
+			Use:     metadata.Aliases[0],
+			Aliases: metadata.Aliases,
+			Short: fmt.Sprintf("Execute %s/%s [%s] extension",
+				metadata.Owner, metadata.Repo, metadata.Tag,
+			),
+			RunE: func(cmd *cli.Command, args []string) error {
+				return execFn(metadata.Aliases[0], args)
+			},
+		}
+		c.DisableFlagParsing = true
+		output[i] = c
+	}
+	return output
+}
+
+func getUsedCommands(cmd *cli.Command) []string {
+	output := make([]string, len(cmd.Commands()))
+	for i, c := range cmd.Commands() {
+		output[i] = c.Name()
+	}
+	return output
+}
+
+func extensionCommand(installFn func(owner, repo, alias string) error, l log.Logger) *cli.Command {
```

Comment:
Can we accept an `interface` here? It will also help write unit tests for the extension package which I don't see any?

Reply:
I don't think `interface` is necessary. We can puth any `installFn` function as we want, since it accepts the pointer. As for the `logger`, I believe it's already an interface.

Reply:
Oh just realised it, for the extension, yes I didn't write it due to schedule. I will add them later and mention it here.

Reply:
I restructured the extension command for better readability, and add interface as well as suggested,